### PR TITLE
KOGITO-5115: Implement integration tests for Save operation & editors…

### PR DESCRIPTION
**Pull request by domhanak**
… in VSCode

Add new BPMN asset to use for save testing.
Add new test case to ensure BPMN editor saves file properly.

cc @jomarko ready for peer-review, I would probably wait for your undo() PR to be merged if possible to add some cool stuff from it, but for general improvements this is ready for review.

### CODE REVIEW

**Comment by jomarko** [comment]
@domhanak sorry, I would like to get firstly in https://github.com/kiegroup/kogito-tooling/pull/473

**Comment by domhanak** [comment]
@jomarko sure, that's what I meant :D

cc @LuboTerifaj  could you please review?

**Comment by domhanak** [comment]
Waiting for #473 to be merged.

File: packages/vscode-extension-pack-kogito-kie-editors/it-tests/settings.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "window.newWindowDimensions": "maximized",
   "workbench.startupEditor": "none",
   "files.simpleDialog.enable": true,
+  "files.autoSave": "onFocusChange",
```
Comment (tiagobento) [inline packages/vscode-extension-pack-kogito-kie-editors/it-tests/settings.json]: Won't that save the file regardless of actions on the editor?

**Comment by jomarko** [comment]
There is one risk/consequence of autosave I think. Quite often we check node names in the tests for example. Then can be confusing  to have asserts, depending on results of other tests. In my 'Undo Test' PR #473 , I rename a node, imagine my test fail for some reason, and the node name remains in a state, that is not expected for the rest of the test suite.

**Comment by domhanak** [comment]
Makes sense, I am just testing different approaches. :)

**Comment by tiagobento** [comment]
@domhanak I marked this PR as WIP, please let us know when this is good for review.

**Comment by domhanak** [comment]
@tiagobento it had the `do not merge` label so i thought it was clear.

rebased on master now and ready for review now.

**Comment by domhanak** [comment]
`CI :: Integration Tests / online-editor-tests-it (windows-latest, 16.2.0, 1.22.10, 4.0.0) (pull_request) Failing after 11m — online-editor-tests-it (windows-latest, 16.2.0, 1.22.10, 4.0.0)` 
is unrelated to these changes

**Review by jomarko — APPROVED** [review]
If we rename just the top level process, maybe I would consider to use smaller asset for such test, without nodes inside. But keep this comment as my point of view, leaving the decision up to you/rest of the team. Thank you.

**Comment by domhanak** [comment]
@jomarko I could extend it to nodes too, however that will make the test significantly longer.

**Comment by jomarko** [comment]
@domhanak yes, I understand and agree. My point was to keep the test as it is, what allows us to use very small (practically empty) bpmn asset.

**Comment by domhanak** [comment]
@jomarko Ok, do you think that 4 nodes are going to make a difference? Locally it does not seem that big.

**Comment by jomarko** [comment]
@domhanak as you can see, I approved already. I mentioned it as an optional thing you can incorporate, or not.

**Comment by domhanak** [comment]
ok, rebased on master and squashed the extra commit to get CI green.

**Review by tiagobento — APPROVED** [review]

**Pull request merged**